### DISCUSSION TITLE
Comparing Peak Performance of Single vs. Multi-Speaker TTS Models: Seeking Insights and References

### DISCUSSION THREAD

**Discussion by ikpark09** [coqui-ai/TTS#2935] [General Q&A]
Hello. First and foremost, I'd like to express my gratitude to everyone who has provided various assistance. Thanks to you, I've been able to progress in my studies, create a VITS TTS model, and synthesize voices.

I have a question about the peak performance of the model based on the training data. By "peak performance," I mean the model's ability to read a variety of texts accurately.

Here's what I've observed during my training:
A single-speaker model trained with 12 hours of Speaker A's data, when fine-tuned with 2 hours of Speaker B's data, seems to have a lower peak performance than the original model trained solely on Speaker A's data. (For fine-tuning, I used different data for B than A. Would the results of fine-tuning be better if I used the same text as A's data?)

Therefore, I believe securing a base model with high peak performance is crucial. While I think it would be beneficial to obtain and train with more high-quality single-speaker data, it has been challenging to acquire additional data in reality.

Suppose we have models trained in the following ways:

1. Single-speaker model1 trained with 10 hours of Speaker A's data.
2. Multi-speaker model2 trained with model1 + 2 hours of Speaker B's data + 2 hours of Speaker C's data.
3. Model3 fine-tuned with model1 + 2 hours of Speaker B's data.

I suspect the order of peak performance would be model2 > model1 > model3. Is this correct? 
I've tried to find clear explanations or evidence, such as papers or articles, for this case but haven't been successful.

My concern is that when training a multi-speaker model, even if there's a lot of overall data, the data for each speaker is separated by speaker IDs. 
So, would the data for each speaker not contribute to the peak performance?

Testing and comparing all these cases would require a lot of resources and time, so I'm seeking assistance here. Does anyone have experience or related materials to share on this topic?

**Comment by pivolan** [comment]
Hi, I have some experiments with this. 
My work was:
1. train single voice de model on thorsten dataset 20hours. 
2. do the same for yourTTS multivoice speaker manager
3. do the same YourTTS multivoice speaker encoder on d_vectors.
this is my base models for finetune in the future to add new voices. I have another voices with 3 speakers: 20 A, 3 hours B, 4 hours C

Much clear voice audio I get when use singlevoice models and finetune. Also I tried train from the scratch speaker C with 4 hours, also very good results, but too muuch errors in pronounciations. 

finetune:
1. finetune single voice: **A 20 minutes**(speech responsible, but unusable, not enough, robovoice, scratch, and many misspronounces), **B 3 hours**, much better audio is very clear on short sentences quality is very good and no error in prnounce, in big texts misspronounciations will be happen often. **C 4 hours**, very good results, but still has misspronounce. 
2. finetune multivoice, with speaker manager. Didn't finished yet. Results was worst then singlevoice. But I still didn't finish with some settings. And was very much errros in pronounce. For added voice speech was absolutely invalid. 
3. For the current time it was a best results. on small datasets finetuning show very good results. experiments: **A 20 minutes** speech responsible, no robovoice, no scratched and no very dirty, also no much missprnounciations. B 3 hours and C 4 hours has similar results, no pronounce errors, but audio much less clear than in single voice model. 

What conclusion I would make: When we use yourTTS multivoice + encoder d_vector pronunciation is taken from the base model, and if the new dataset lacks data how to pronounce something, it will be taken from the base model, and then with the help of encoder will correct the sound of the voice, at the same time pauses and tempo of speech are also copied. It's as if the vocoder is being trained more and the melgram model is almost unchanged.
In case of single voice training, pronunciation is often broken.
Also, if you have 4-5 hours of audio, it makes sense to try to train from scratch, I got the cleanest sound, but a lot of errors in pronunciation.
